### PR TITLE
fix #7502: write run_results.json for run operation

### LINKED ISSUE #7502: [Feature] `run-operation` should write a run result artifact which should include `args`
## Description
While looking into `dbt retry` #7299, it was found that the `run-operation` command should also retryable. The requirement for a command to be retryable is that they produce a `run_results.json` artifact that includes an `args` field containing all params used in the invoked command and that this `args` field include a `which` attribute with the name of the command.

## Acceptance Criteria
- The `RunOperationResult` class includes an `args` field
  - This `args` field includes a `which` attribute set to the string `"run-operation"`
- The `run-operation` command should write `run_results.json`

## Notes
- [Example of `RunExecutionResult` accepting args allowing itself to be writable](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/contracts/results.py#L205-L220)
- [`GraphRunnableTask` has a write_result](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/task/runnable.py#L411-L412) method that could help here, but `RunOperationTask` does not inherit from it. It may be a good idea to either move this `write_result` method to the `ConfiguredTask` class so `RunOperationTask` can inherit from it or abstract it from the task structure entirely
- As per @MichelleArk, running `scripts/collect-artifact-schema.py --path schemas` should update/create new artifacts as necessary, and you can git restore the non run-result artifacts afterwards since they only have some dates updated, though if a breaking change is made the schema version may need to be bumped manually

### PR DESCRIPTION
**Pull request by aranke**
resolves #7502 

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Review by ChenyuLInx — CHANGES_REQUESTED** [review]
Thanks for adding this! Looks mostly good to me with a few questions. I hope we can avoid abstract one liner that only being used once into a function. I understand that some of it might be just following what other places did, but existing code have certain bad patterns that made reading the code a lot harder than needs, we should try to correct them when encounter.

File: tests/functional/artifacts/test_artifacts.py [hunk 1]
```diff
@@ -649,3 +670,28 @@ def test_versions(self, project, manifest_schema_path, run_results_schema_path):
...
+        }
+
+    def test_run_operation(self, project):
+        results, log_output = run_dbt_and_capture(["run-operation", "alter_timezone"])
+        assert len(results) == 1
+        assert results[0].status == RunStatus.Success
+        assert results[0].unique_id == "operation.test.alter_timezone"
+        assert "Timezone set to: America/Los_Angeles" in log_output
+
+    def test_run_model(self, project):
```
[the remark this answers is not in the document]
Comment (ChenyuLInx) [reply tests/functional/artifacts/test_artifacts.py:693]: Leaving the test makes sense! Can we rename the test to something like run model with operation and add a bit more comment about why run operation should not be in results in this case?

**Review by ChenyuLInx — APPROVED** [review]
Thanks for adding this patch! From my perspective the rest of the comment are optional for this PR. Feel free to merge as is or add those then merge.

File: tests/functional/artifacts/test_artifacts.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jtcohen6) [reply tests/functional/artifacts/test_artifacts.py]: Discussed live: Macros run as pre/post hooks on a single model aren't recorded in `run_results.json` — that tracks for me, given how the run results artifact works today. (Run results also don't account for multi-step materializations.)

**Pull request merged**